### PR TITLE
feat: finished form builder page

### PR DESCRIPTION
**Pull request by artikandri**
## What's changed: 
- ui/.eslintrc.js: added new rules
- ui/package.json: added new packages: vuedraggable to support dragging items and vuetify-notify to support calling a function to show notification from the script block instead of declaring snackbar every single time 
- ui/src/App.vue: added id="app"
- ui/src/api/index.js: added interceptors to handle api response/request by default. **NOT TESTED YET.**
- ui/src/components/BottomSheet.vue: refactored the component, added additional props 
- ui/src/components/Modal.vue: refactored the component; added additional props
- ui/vue.config.js: added an option to allow vue to compile on runtime; might be discarded later. 
- ui/src/styles/overrides.scss : added some new styles to override basic vuetify styling
- layouts/* : deleted unused components
- user/settings: removed material snackbar and replaced with notify.toast
- admin/settings: removed material snackbar and replaced with notify.toast

## What's new: 
- ui/src/assets/json/form-builder-list-sample.json: contains a sample of generated form builder component list 
- ui/src/form-builder/* : Files related to FormBuilder component. Put this outside the views because it may get imported by other views (ex: generateForm component will be imported in survey detail and respondent survey pages)
- ui/src/form-builder/components/components.js: contains basic json structure for the component
- ui/src/views/user/survey-edit/* : Files related to the Edit survey page 
- ui/src/plugins/vuedraggable.js, vuetify-notify: New plugins to allow dragging and notification
- ui/src/util/ : added new utilities; notably copyText (to allow copying text to clipboard; event bus (to enable communications between components) 

The rest of the changes were caused by `yarn lint --fix`

### Desktop: 

empty state: 
image
image
image

filled state: 
image

on widget selection: 
image

on widget settings modal view: 
image

on survey: public settings: 
image

on survey: private settings
image

preview page: 
image

### Mobile: 
empty state: 
image

filled: 
image

add new survey element: 
image

settings: 
image

form preview: 
image

on save button click: 
image

## Notable: 
- Added a new function/button that is not in the original design to clear the form
image
- As for now (10/6/2022) I havent implemented the "change question type" feature, because it is prone of bugs
- The mobile design ended up a bit different because I'm too lazy to update the AppBar component; the save button should've been in the app bar 
- I used a different icon for "send invitations" button because I forgot to export the icon from figma
- the original design has the "assistive text" and "required" fields but I didnt implement it. 
- refactored the modal component
- added a function to add widget item by doubleclicking instead of only by dragging in desktop mode
- added a button to enable cloning widget item 

## To do: 
- survey detail page
- validation on survey data. Right now we can submit the form without filling anything, and we will still fulfill the requirement because the requirement doesnt mention anything about validation
- Refactor WidgetConfig component. Right now the form data is updated by using watchers, data is not generated automatically and using watchers to watch every single change is heavy
- major refactor on WidgetFormItem component and GenerateFormItem component. The code can be made cleaner by using dynamic component 
- standardizing component names; now Im using "Settings" and "Config", "Elements" and "Widgets" at the same time. 
- bugs on widgetform item. Sometimes the widget will be added twice and no idea why because the function is only fired once. Now the bug is handled by filtering the widgets before submission, but the real error should be handled first
- Some of the components mutate props directly. This approach is dirty and generally is a bad practice, but I will leave it for now because the survey detail page needs more attention.

### CODE REVIEW

File: ui/src/assets/json/form-builder-list-sample.json
Comment (teddy-f-47) [inline ui/src/assets/json/form-builder-list-sample.json]: The current db design does not distinguish between "short open answer" and "long open answer", so the back-end and db guys should add one column in the open answer class/table like a boolean "isLongOpenAnswer"; for the "options" field i am not sure but maybe they can do some tricks like putting the defaultValue in the answer field right away. Please communicate these 2 needs (distinguish short and long open answers, options) with the back-end and db guys. Please, communicate this with the rest of the team.

**Review by teddy-f-47 — APPROVED** [review]
Approved for merging, but please communicate the json needs with the back-end and db guys. Details in the comment.

File: ui/src/assets/json/form-builder-list-sample.json
Comment (artikandri) [reply ui/src/assets/json/form-builder-list-sample.json]: Lets wait for the others until I merge this MR; just in case I need to delete the textarea XD

Comment (teddy-f-47) [reply ui/src/assets/json/form-builder-list-sample.json]: Based on the recent team discussion, please remove the "long open answer" from ui and json structure. We will only use "short open answer" and call it simply "open answer". Thanks a lot!

Comment (artikandri) [reply ui/src/assets/json/form-builder-list-sample.json]: Alright, I will work on it and notify you later when I finished.

Comment (artikandri) [reply ui/src/assets/json/form-builder-list-sample.json]: image
 Updated.

**Review by teddy-f-47 — APPROVED** [review]
Thank you very much for your hard work! So, we now only have one Open Answer question type. Approved for merging.

**Pull request merged**
